andela/ah-robotics-frontend
# #162949036 Password Reset Link

#### What does this PR do?
Presents a user with a link to reset password
#### Description of Task to be completed?
#### How should this be manually tested?
1. Navigate to a folder where you would prefer to test the repository.

2. Create a directory
 ```$```  ```mkdir myproject```

3. Navigate to the directory "myproject"
 ```$```   ```cd myproject```
4. Clone the authors haven repository
  ```$```   ```git clone https://github.com/andela/ah-robotics-frontend.git```
5. Change the directory into the "ah-robotics-fronted"
  ```$```   ```cd ah-robotics-frontend```
6.  Checkout to the branch ft-reset-password-link-162949036
 ```$```   ```git checkout ft-reset-password-link-162949036```
7. Install dependancies from the package.json
 ```$```   ```npm install``` to install all the modules listed in the `package.json` file.

8. Start the application in development mode
  ```$```   ```npm start``` to run the app in the development mode 

The application opens http://localhost:3000 in the browser.
Change the URL on the browser to;
 ```https://ah-robotics-frontend-sta-pr-11.herokuapp.com/resetpassword```

The browser will show like represented in screenshot _**image: 1**_

9. Enter a valid email for demo purpose use;
```roboticsah@gmail.com``` and click on the ```ResetPassword``` button.
You will get a status prompt if success or not. See _**image: 2**_.

10. Go to the email box used and check the reset message sent. See _**Image: 3**_

11. Click on the page ```Click here``` highlighted at the center of the email message.

You get redirected to a screen to with a *change password* and a *confirm password* See _**Image: 4**_

12. Enter the _new password_ and _confirm password_ and click ```Reset Password``` button.
You will get a success or failure message. If success check your email for confrimation message. See _**Image: 5**_ 

13. To run tests, open the terminal and run the following command:
See **_Image: 6_**
```
npm test
``` 
14. To run tests and coverage, open the terminal and run the following command:
```
npm test -- --coverage
```
<img width="978" alt="screenshot 2019-02-15 at 09 02 27" src="https://user-images.githubusercontent.com/39020475/52837637-97ea8180-3100-11e9-8c09-7499ce576d1a.png">

Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
[#162949036](https://www.pivotaltracker.com/story/show/162949036)
#### Screenshots (if appropriate)

##### Image:1 Password reset page

<img width="1246" alt="screenshot 2019-02-14 at 15 54 25" src="https://user-images.githubusercontent.com/39020475/52788098-ef411100-3070-11e9-8952-99c6b30e49f7.png">

##### Image: 2 Success message
<img width="1243" alt="screenshot 2019-02-14 at 16 10 42" src="https://user-images.githubusercontent.com/39020475/52788882-61b2f080-3073-11e9-8e7a-3c7420a24455.png">

##### Image: 3 Email message with reset password prompt
<img width="995" alt="screenshot 2019-02-14 at 16 08 02" src="https://user-images.githubusercontent.com/39020475/52789006-bce4e300-3073-11e9-8caf-ddfc261be610.png">

##### Image: 4 Password change and confirm interface

![image 1](https://user-images.githubusercontent.com/39020475/52790375-71343880-3077-11e9-900d-6346a60edc9f.png)

##### Image: 5 Password change email confirmation message.

![image 2](https://user-images.githubusercontent.com/39020475/52790714-4c8c9080-3078-11e9-8a68-5ba57ec91bad.png)

##### Image: 6 Test run on branch
<img width="750" alt="screenshot 2019-02-14 at 21 37 37" src="https://user-images.githubusercontent.com/39020475/52810533-d43bc480-30a3-11e9-9bff-15bc3610cdd7.png">
.


#### Questions:
